singer_sdk/plugin_base.py

```diff
@@ -48,7 +50,11 @@
 class PluginBase(metaclass=abc.ABCMeta):
     """Abstract base class for taps."""
 
-    name: str  # The executable name of the tap or target plugin.
+    #: The executable name of the plugin. e.g. tap-csv
+    name: str
+
+    #: The package name of the plugin. e.g. meltanolabs-tap-csv
+    package_name: str | None = None
```

Comment:
> Also - I'm confused why this worked for other taps, since the `tap_id` in the cookiecutter has dashes (`'tap-snowflake'`) and the package name should have those converted to underscores (`'tap_snowflake'`). Is the python internal libraries auto-converting the dashes to underscores to make it work?

@aaronsteers `importlib.metadata.version` _normalizes_ the package name according to [PEP 440](https://peps.python.org/pep-0440/), so

```python
importlib.metadata.version("tap_snowflake")

# returns the same value as
importlib.metadata.version("tap-snowflake")
```

Reply:
@aaronsteers the code that fetches the version ([here](https://github.com/meltano/sdk/blob/main/singer_sdk/plugin_base.py#L168)) actually requires the package name form (same as the wheel name or the pip install name) which is `meltanolabs-tap-snowflake` in my case. So `--about` will only correctly display a version when the `tap_id`/`PluginBase.name` matches the install package name. That is probably why this just works in most cases, and why it hasn't been raised before - you don't get any hard error when the names are mismatched - you just get the string ` Version v[could not be detected]` appear in `--about` (I noticed it [here](https://github.com/meltano/hub/pull/1061#issuecomment-1337030831) in the "Version info" section of the meltybot comment on my hub PR).

However `PluginBase.name` is used both for logging and as a stream-name prefix. I think that i) it is neater for logs to match the invocation name (i.e. if I invoke `tap-snowflake` I expect the logs to use that name and not the pip/wheel name I happened to use for install, and in the Meltano context I may never have even seen if I installed via `meltano add extractor tap-snowflake --variant meltanolabs`) and ii) it is useful for different variants to output the same streams (to make moving from one variant of `tap-snowflake` to another as seamless as possible).

This is how the PPW variants work; the `pipelinewise-` prefix is only used for installation - e.g. [pipelinewise-tap-snowflake](https://github.com/transferwise/pipelinewise-tap-snowflake/blob/master/setup.py?rgh-link-date=2022-12-05T19%3A15%3A36Z) is imported as `tap_snowflake` and executed as `tap-snowflake` 🙂 We don't have to follow the same convention (we could use `meltanolabs-tap-snowflake` and the underscore variation as the tap_id, library name and invocation/entrypoint name) but I do really like how the PPW ones work 😅

Reply:
> I wonder now why `tap_snowflake` doesn't also report a version 🤔

@kgpayne I don't think it should, really. The `tap_snowflake` directory name is only relevant for imports, but not for _distrubution_. Think of [`python-dateutil`](https://dateutil.readthedocs.io/en/stable/), installed as such, but imported as `dateutil` and living under `src/dateutil` in the source distribution.

Reply:
@aaronsteers I think that would only help to detect the top-level directory name of the package, so in the case of the `dateutil` example above it wouldn't be able to get the `python-dateutil` name from the module path _when the package is installed_.